indico/htdocs/js/indico/modules/registration/form/field.js

```diff
@@ -674,12 +688,189 @@ ndRegForm.directive('ndEmailField', function(url) {
     };
 });
 
+ndRegForm.directive('ndAccommodationField', function(url) {
+    return {
+        require: 'ndField',
+        controller: function($scope) {
+            $scope.tplInput = url.tpl('fields/accommodation.tpl.html');
+            $scope.areArrivalDatesValid = function(data) {
+                return moment(data['arrivalDateTo'], "DD-MM-YYYY")
+                        .isAfter(moment(data['arrivalDateFrom'], "DD-MM-YYYY"));
+            };
+
+            $scope.areDepartureDatesValid = function(data) {
+                return moment(data['departureDateTo'], "DD-MM-YYYY")
+                        .isAfter(moment(data['departureDateFrom'], "DD-MM-YYYY"))
+            };
+
+            $scope.areAccommodationOptionsDefined = function(data) {
+                var choices = data.choices.filter(function(item) {
+                    return item.remove !== true;
+                });
+                return choices.length !== 0;
+            }
+
+            $scope.validateFieldSettings = function(dialogScope) {
+                var data = dialogScope.formData;
+                if (!$scope.areArrivalDatesValid(data)) {
+                    dialogScope.$apply(dialogScope.setSelectedTab('tab-accommodation'));
+                    return false;
+                }
+
+                if (!$scope.areDepartureDatesValid(data)) {
+                    dialogScope.$apply(dialogScope.setSelectedTab('tab-accommodation'));
+                    return false;
+                }
+
+                if (!$scope.areAccommodationOptionsDefined(data)) {
+                    dialogScope.$apply(dialogScope.setSelectedTab('tab-accommodation-options'));
+                    return false;
+                }
+
+                return true;
+            };
+
+            $scope.initFormData = function(formData, field) {
+                var eventSDate = $scope.eventStartDate
```

Comment:
`startDate` and `endDate` would be much clearer when reading this code instead of `eventSDate` and `eventEDate`